### PR TITLE
Series Admin User Interface

### PR DESCRIPTION
**Pull request by bordoni**
_Ref:_ [C#70879](https://central.tri.be/issues/70879)

### CODE REVIEW

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:x: | Logical operator &quot;or&quot; is prohibited; use &quot;||&quot; instead | [Template.php:3](https://github.com/moderntribe/tribe-common/blob/d0492d321fdb3a8da786dfc6fb6033ec08577aa8/src/Tribe/Admin/Template.php#L3)
:x: | Extract is the devil. Use separate assignments or access the array directly instead. | [Template.php:278](https://github.com/moderntribe/tribe-common/blob/d0492d321fdb3a8da786dfc6fb6033ec08577aa8/src/Tribe/Admin/Template.php#L278)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$html' | [Template.php:304](https://github.com/moderntribe/tribe-common/blob/d0492d321fdb3a8da786dfc6fb6033ec08577aa8/src/Tribe/Admin/Template.php#L304)

> via jenkins/codesniffer-phpcs

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:x: | Logical operator &quot;or&quot; is prohibited; use &quot;||&quot; instead | [Template.php:3](https://github.com/moderntribe/tribe-common/blob/d1a2621b970bf0e49b0ddfc1911a44e4c75f41c8/src/Tribe/Template.php#L3)
:x: | Extract is the devil. Use separate assignments or access the array directly instead. | [Template.php:289](https://github.com/moderntribe/tribe-common/blob/d1a2621b970bf0e49b0ddfc1911a44e4c75f41c8/src/Tribe/Template.php#L289)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$html' | [Template.php:315](https://github.com/moderntribe/tribe-common/blob/d1a2621b970bf0e49b0ddfc1911a44e4c75f41c8/src/Tribe/Template.php#L315)
:x: | Echoed strings should not be bracketed | [tabbed-view.php:23](https://github.com/moderntribe/tribe-common/blob/d1a2621b970bf0e49b0ddfc1911a44e4c75f41c8/src/admin-views/tabbed-view/tabbed-view.php#L23)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$tab' | [tabbed-view.php:23](https://github.com/moderntribe/tribe-common/blob/d1a2621b970bf0e49b0ddfc1911a44e4c75f41c8/src/admin-views/tabbed-view/tabbed-view.php#L23)
:x: | Echoed strings should not be bracketed | [tabbed-view.php:25](https://github.com/moderntribe/tribe-common/blob/d1a2621b970bf0e49b0ddfc1911a44e4c75f41c8/src/admin-views/tabbed-view/tabbed-view.php#L25)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '!' | [tabbed-view.php:25](https://github.com/moderntribe/tribe-common/blob/d1a2621b970bf0e49b0ddfc1911a44e4c75f41c8/src/admin-views/tabbed-view/tabbed-view.php#L25)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not ':' | [tabbed-view.php:25](https://github.com/moderntribe/tribe-common/blob/d1a2621b970bf0e49b0ddfc1911a44e4c75f41c8/src/admin-views/tabbed-view/tabbed-view.php#L25)

> via jenkins/codesniffer-phpcs

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:x: | Logical operator &quot;or&quot; is prohibited; use &quot;||&quot; instead | [Template.php:3](https://github.com/moderntribe/tribe-common/blob/a9714909952bca2eadd9face2d0779674594d0ad/src/Tribe/Template.php#L3)
:x: | Extract is the devil. Use separate assignments or access the array directly instead. | [Template.php:289](https://github.com/moderntribe/tribe-common/blob/a9714909952bca2eadd9face2d0779674594d0ad/src/Tribe/Template.php#L289)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$html' | [Template.php:315](https://github.com/moderntribe/tribe-common/blob/a9714909952bca2eadd9face2d0779674594d0ad/src/Tribe/Template.php#L315)
:x: | Echoed strings should not be bracketed | [tabbed-view.php:23](https://github.com/moderntribe/tribe-common/blob/a9714909952bca2eadd9face2d0779674594d0ad/src/admin-views/tabbed-view/tabbed-view.php#L23)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$tab' | [tabbed-view.php:23](https://github.com/moderntribe/tribe-common/blob/a9714909952bca2eadd9face2d0779674594d0ad/src/admin-views/tabbed-view/tabbed-view.php#L23)
:x: | Echoed strings should not be bracketed | [tabbed-view.php:25](https://github.com/moderntribe/tribe-common/blob/a9714909952bca2eadd9face2d0779674594d0ad/src/admin-views/tabbed-view/tabbed-view.php#L25)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '!' | [tabbed-view.php:25](https://github.com/moderntribe/tribe-common/blob/a9714909952bca2eadd9face2d0779674594d0ad/src/admin-views/tabbed-view/tabbed-view.php#L25)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not ':' | [tabbed-view.php:25](https://github.com/moderntribe/tribe-common/blob/a9714909952bca2eadd9face2d0779674594d0ad/src/admin-views/tabbed-view/tabbed-view.php#L25)

> via jenkins/codesniffer-phpcs

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:x: | Logical operator &quot;or&quot; is prohibited; use &quot;||&quot; instead | [Template.php:3](https://github.com/moderntribe/tribe-common/blob/f834f0be795e6983e7d81725c6c2aa82b63e4643/src/Tribe/Template.php#L3)
:x: | Extract is the devil. Use separate assignments or access the array directly instead. | [Template.php:289](https://github.com/moderntribe/tribe-common/blob/f834f0be795e6983e7d81725c6c2aa82b63e4643/src/Tribe/Template.php#L289)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$html' | [Template.php:315](https://github.com/moderntribe/tribe-common/blob/f834f0be795e6983e7d81725c6c2aa82b63e4643/src/Tribe/Template.php#L315)
:x: | Echoed strings should not be bracketed | [tabbed-view.php:23](https://github.com/moderntribe/tribe-common/blob/f834f0be795e6983e7d81725c6c2aa82b63e4643/src/admin-views/tabbed-view/tabbed-view.php#L23)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$tab' | [tabbed-view.php:23](https://github.com/moderntribe/tribe-common/blob/f834f0be795e6983e7d81725c6c2aa82b63e4643/src/admin-views/tabbed-view/tabbed-view.php#L23)
:x: | Echoed strings should not be bracketed | [tabbed-view.php:25](https://github.com/moderntribe/tribe-common/blob/f834f0be795e6983e7d81725c6c2aa82b63e4643/src/admin-views/tabbed-view/tabbed-view.php#L25)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '!' | [tabbed-view.php:25](https://github.com/moderntribe/tribe-common/blob/f834f0be795e6983e7d81725c6c2aa82b63e4643/src/admin-views/tabbed-view/tabbed-view.php#L25)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not ':' | [tabbed-view.php:25](https://github.com/moderntribe/tribe-common/blob/f834f0be795e6983e7d81725c6c2aa82b63e4643/src/admin-views/tabbed-view/tabbed-view.php#L25)

> via jenkins/codesniffer-phpcs

File: src/Tribe/Template.php [hunk 1]
```diff
@@ -0,0 +1,320 @@
+<?php
+// Don't load directly
+defined( 'WPINC' ) or die;
```
Comment (lucatume) [inline src/Tribe/Template.php]: As a note: this prevents any autoloading not in a WordPress context to happen. While I know this is common WordPress practice one has to keep that in mind.

File: src/Tribe/Template.php [hunk 2]
```diff
@@ -0,0 +1,320 @@
...
+
+	/**
+	 * Configures the class origin plugin path
+	 *
+	 * @param  object|string  $origin   The base origin for the templates
+	 *
+	 * @return self
+	 */
+	public function set_template_origin( $origin = null ) {
+		if ( ! isset( $origin ) ) {
```
Comment (lucatume) [inline src/Tribe/Template.php]: `empty($origin)`? So that a `''` or `false` origin case is handled too?

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:x: | Logical operator &quot;or&quot; is prohibited; use &quot;||&quot; instead | [Template.php:3](https://github.com/moderntribe/tribe-common/blob/abf5e40b7346584aa5cb25d9a15cf78dd89ca568/src/Tribe/Template.php#L3)
:x: | Extract is the devil. Use separate assignments or access the array directly instead. | [Template.php:289](https://github.com/moderntribe/tribe-common/blob/abf5e40b7346584aa5cb25d9a15cf78dd89ca568/src/Tribe/Template.php#L289)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$html' | [Template.php:315](https://github.com/moderntribe/tribe-common/blob/abf5e40b7346584aa5cb25d9a15cf78dd89ca568/src/Tribe/Template.php#L315)
:x: | Echoed strings should not be bracketed | [tabbed-view.php:23](https://github.com/moderntribe/tribe-common/blob/abf5e40b7346584aa5cb25d9a15cf78dd89ca568/src/admin-views/tabbed-view/tabbed-view.php#L23)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$tab' | [tabbed-view.php:23](https://github.com/moderntribe/tribe-common/blob/abf5e40b7346584aa5cb25d9a15cf78dd89ca568/src/admin-views/tabbed-view/tabbed-view.php#L23)
:x: | Echoed strings should not be bracketed | [tabbed-view.php:25](https://github.com/moderntribe/tribe-common/blob/abf5e40b7346584aa5cb25d9a15cf78dd89ca568/src/admin-views/tabbed-view/tabbed-view.php#L25)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '!' | [tabbed-view.php:25](https://github.com/moderntribe/tribe-common/blob/abf5e40b7346584aa5cb25d9a15cf78dd89ca568/src/admin-views/tabbed-view/tabbed-view.php#L25)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not ':' | [tabbed-view.php:25](https://github.com/moderntribe/tribe-common/blob/abf5e40b7346584aa5cb25d9a15cf78dd89ca568/src/admin-views/tabbed-view/tabbed-view.php#L25)

> via jenkins/codesniffer-phpcs

**Pull request merged**